dfinity/nns-dapp
# GIX-1267: Use "query" not signed in users

# Motivation

Not logged in users see uncertified data.

This reduces the load of update calls.

# Changes

Main change
* queryAndUpdate can't be used with other than "query" when identity is anonymous, it raises an error otherwise.

Minor change
* Show the test error when loading sns nervous functions if call fails and not certified but it's anonymous identity.

# Tests

* Add two new workflow tests to check that a user load different data in NnsProposals and NnsProposalDetails.
* Rename the file where queryAndUpdate was being tested to the same where it lives.
* Add a test case in queryAndUpdate test.
* Fix tests after change in functionality.
